src/components/course/routes/tests/ExternalCourseEnrollment.test.jsx

```diff
@@ -136,6 +147,36 @@ describe('ExternalCourseEnrollment', () => {
     expect(screen.getByTestId('user-enrollment-form')).toBeInTheDocument();
   });
 
+  it.each([
+    // The course exists and is "available", but non-redeemable via subsidy.
+    {
+      contentKey: mockCourseRunKey,
+      hasSuccessfulRedemption: false,
+      canRedeem: false,
+    },
+    // The requested course run is not included in the can-redeem response, so it is "unavailable".
+    {
+      contentKey: 'some+other+course', // mockCourseRunKey represents the run requested in the URL.
+      hasSuccessfulRedemption: false,
+      canRedeem: false,
+    },
+  ])('renders 404 page when appropriate', (mockCanRedeemData) => {
```

Comment:
[nit] The test cases don't really handle if `canRedeem` is true. A quick update to the test could be conditionally tested on canRedeem
```js
if(!canRedeem){
  expect(screen.getByTestId('not-found-page')).not.toBeInTheDocument();
} else {
  expect(screen.getByTestId('not-found-page')).toBeInTheDocument();
}
```

Reply:
The default mock return for canRedeem is `true` on line 122, so all other tests are effectively testing the canRedeem=true case. Does that seem sufficient?  I think I could be more explicit by setting canRedeem to true on line 161 to prove that we *still* get 404 even if other course is redeemable.

Reply:
the logic got more complicated, expanded both the happy case and 404 test cases to be data-driven.